pkgs/shells/bash/5.nix

```diff
@@ -133,10 +133,26 @@ lib.warnIf (withDocs != null)
 
     enableParallelBuilding = true;
 
-    makeFlags = lib.optionals stdenv.hostPlatform.isCygwin [
-      "LOCAL_LDFLAGS=-Wl,--export-all,--out-implib,libbash.dll.a"
-      "SHOBJ_LIBS=-lbash"
-    ];
+    makeFlags =
+      lib.optionals stdenv.hostPlatform.isCygwin [
+        "LOCAL_LDFLAGS=-Wl,--export-all,--out-implib,libbash.dll.a"
+        "SHOBJ_LIBS=-lbash"
+      ]
+      # GNU Bash extensively uses function prototypes with unspecified signature, like
+      #
+      #     extern void add_unwind_protect ()
+      #
+      # to implicitly cast all kinds of word-sized arguments. GCC-15 switched C standard
+      # from gnu17 to gnu23, which changed the meaning of the prototype above to
+      #
+      #     extern void add_unwind_protect (void)
+      #
+      # Given that GNU Bash is committed to support pre-standard compilers, I don't know
+      # how to produce an upstreamable patch for this situation. For now, just downgrade
+      # the C standard.
+      #
+      #      ~kaction 2025-06-28
+      ++ [ "LOCAL_CFLAGS=-std=c99" ];
```

Comment:
Wasn't the plan here to upgrade Bash?

If this too much of an extra hassle now, that's OK. Please mention in your comment that this can go with Bash 5.3 though as this doesn't reflect the reality IMHO.

Reply:
Yes, the plan is to upgrade Bash, but I don't know when I will be able to allocate the time for that. 
For now, I want to focus on merging things already implemented, so they don't get stale and merge-conflicted.

Updated the comment about bash, mentioning that upstream resolved the issue is bash=5.3.

Reply:
All good, just wanted to make sure this didn't slip :)